ExampleApp/ExampleApp/Common/DataSource/TableViewDataSource.swift

```diff
@@ -0,0 +1,83 @@
+import RxSwift
+import RxBluetoothKit
+import UIKit
+
+/* Generic UITableViewDataSource. I - represents model type, S - represents SectionModelItem, which stores
+   information about given section's itemsCount, sectionName, cells' class and provides collection of model data
+   check: DataSource.SectionModelItem.swift
+*/
+
+class TableViewDataSource<I, S:SectionModelItem>: NSObject, UITableViewDataSource {
+
+    // MARK: - Typealiases
+    // Block used to configure UITableViewCell, passed into init
+    typealias CellConfigurationBlock = (_ cell: UITableViewCell, _ item: Any) -> Void
+
+    // Block set by given UIViewController, meant to be called for reloading data
+    typealias RefreshDataBlock = () -> Void
+
+    // Block set by given UIViewController, meant to be called inside at any OnError
+    typealias OnErrorBlock = (_ error: Error) -> Void
+
+    // MARK: - Fields
+    var refreshDataBlock: RefreshDataBlock?
+
+    var onErrorBlock: OnErrorBlock?
```

Comment:
Also: since the signature of this property is void returning, maybe you could provide the default no-op `{ _ in }` implementation instead of wrapping it into the optional?

Reply:
Done